src/components/form/form.js

```diff
@@ -0,0 +1,146 @@
+import React, { Component } from 'react';
+
+import { initialState } from './initial-state';
+import './form.css';
+
+export default class Form extends Component {
+  
+  state = {
+    firstName: "",
+    lastName: "",
+    phone: "",
+    email: "",
+    emailError: "",
+    emailErrorMessage: ""
+  }
+
+  getInitialState = () => {
+    this.setState(initialState)
+  };
+
+  handleChange = (e) => {
+    this.setState({
+      [e.target.name]: e.target.value
+    })
+  };
+
+  // email validation
+
+  validate = () => {
+    const { email } = this.state;
+    if (!email) {
+      this.setState({
+        emailError: "invalid",
+        emailErrorMessage: "Please enter email"
+      })
+      return false
```

Comment:
can we use it in state, instead of returning value.

Reply:
Unfortunately I can't understand what do you mean. Need some details.
We can do it in a few ways. So, give me please extra information and I will fix it.